util.py

```diff
@@ -360,7 +360,16 @@ def get_tags(agentConfig):
             region = instance_identity['region']
 
             import boto.ec2
-            connection = boto.ec2.connect_to_region(region, aws_access_key_id=iam_params['AccessKeyId'], aws_secret_access_key=iam_params['SecretAccessKey'], security_token=iam_params['Token'])
+            proxy_settings = get_proxy(agentConfig)
+            connection = boto.ec2.connect_to_region(
+                region,
+                aws_access_key_id=iam_params['AccessKeyId'],
+                aws_secret_access_key=iam_params['SecretAccessKey'],
+                security_token=iam_params['Token'],
+                proxy=proxy_settings['host'], proxy_port=proxy_settings['port'],
```

Comment:
If there's no proxy configured won't `get_proxy()` return None, and then you'd have a `TypeError` here?